File: BeatPlannerAndroid/src/beatplanner/ui/MetroActivity.cs
﻿
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Android.App;
using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Views;
using Android.Widget;
using BeatPlanner;
using Android.Media;
using Android.Util;

namespace BeatPlannerAndroid
{
  [Activity(Label = "Metronome")]
  public class MetroActivity : Activity
  {
    private Metronome metro;
    private SoundPool sp;
    private int claveId, claveLowId;

    Button startBtn;
    Button stopBtn;
    EditText bpmText;
    SeekBar bpmSeekBar;
    Button bpmIncBtn;
    Button bpmDecBtn;

    Spinner lowerMeterSpinner;

    Spinner upperMeterSpinner;

    protected override void OnCreate(Bundle bundle)
    {
      base.OnCreate(bundle);
      SetContentView(Resource.Layout.Metro);
      Init();
    }

    private void Init()
    {

      metro = new Metronome();
      sp = new SoundPool(2, Android.Media.Stream.Music, 0);
      claveId = sp.Load(this, Resource.Raw.llfclave, 1);
      claveLowId = sp.Load(this, Resource.Raw.llfclave_low, 1);
      metro.BeatEvent += (object sender, BeatEventArgs e) =>
      {
        int sid;
        if (e.Index == 1)
          sid = claveId;
        else
          sid = claveLowId;
        sp.Play(sid, 1.0f, 1.0f, 1, 0, 1.0f);
      };
      metro.BPMChangedEvent += OnBPMChanged;

      // UI init
      startBtn = FindViewById<Button>(Resource.Id.startBtn);
      stopBtn = FindViewById<Button>(Resource.Id.stopBtn);
      bpmText = FindViewById<EditText>(Resource.Id.bpmText);
      bpmSeekBar = FindViewById<SeekBar>(Resource.Id.bpmSeekBar);
      bpmIncBtn = FindViewById<Button>(Resource.Id.inc_tempo);
      bpmDecBtn = FindViewById<Button>(Resource.Id.dec_tempo);
      upperMeterSpinner = FindViewById<Spinner>(Resource.Id.upperMeter);
      lowerMeterSpinner = FindViewById<Spinner>(Resource.Id.lowerMeter);

      var upperAdapter = ArrayAdapter.CreateFromResource(
                           this,
                           Resource.Array.upperMeterValues, 
                           Android.Resource.Layout.SimpleSpinnerItem
                         );

      upperAdapter.SetDropDownViewResource(Android.Resource.Layout.SimpleSpinnerDropDownItem);
      upperMeterSpinner.Adapter = upperAdapter;

      var lowerAdapter = ArrayAdapter.CreateFromResource(
                           this,
                           Resource.Array.lowerMeterValues, 
                           Android.Resource.Layout.SimpleSpinnerItem
                         );

      lowerAdapter.SetDropDownViewResource(Android.Resource.Layout.SimpleSpinnerDropDownItem);
      lowerMeterSpinner.Adapter = lowerAdapter;

      // Initialize UI with values
      bpmText.Text = metro.BPM.ToString();
      bpmSeekBar.Progress = metro.BPM;
      upperMeterSpinner.SetSelection(3);
      lowerMeterSpinner.SetSelection(1);

      upperMeterSpinner.ItemSelected += 
        (object sender, AdapterView.ItemSelectedEventArgs e) =>
      {
        metro.Beat.Meter.Upper = 
          Convert.ToInt32(e.Parent.GetItemAtPosition(e.Position).ToString());
      };

      lowerMeterSpinner.ItemSelected += 
        (object sender, AdapterView.ItemSelectedEventArgs e) =>
      {
        metro.Beat.Meter.Lower = 
          Convert.ToInt32(e.Parent.GetItemAtPosition(e.Position).ToString());
      };

      startBtn.Click += (object sender, EventArgs e) => metro.Start();
      stopBtn.Click += (object sender, EventArgs e) => metro.Stop();

      bpmIncBtn.Click += (object sender, EventArgs e) => metro.BPM += 5;


      bpmDecBtn.Click += (object sender, EventArgs e) => metro.BPM -= 5;

      bpmText.FocusChange += (object sender, View.FocusChangeEventArgs e) =>
      {
        try
        {
          if (e.HasFocus == false)
            metro.BPM = Convert.ToInt32(bpmText.Text);
        }
        catch (FormatException ex)
        {
          Log.Error(Class.SimpleName, ex.StackTrace);
        }
      };

      bpmSeekBar.ProgressChanged += 
        (object sender, SeekBar.ProgressChangedEventArgs e) => 
          metro.BPM = e.Progress;

    }

    void OnBPMChanged(object sender, BPMChangedEventArgs e)
    {
      bpmText.Text = e.BPM.ToString();
      bpmSeekBar.Progress = e.BPM;
    }
  }
}

File: app/src/main/java/dk/aschoen/beatplanner/ui/activities/MetroActivity.java
//
// Translated by CS2J (http://www.cs2j.com): 2014-07-29 22:02:12
//

package dk.aschoen.beatplanner.ui.activities;


import android.app.Activity;
import android.media.SoundPool;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.Button;
import android.widget.EditText;
import android.widget.SeekBar;
import android.widget.Spinner;

import dk.aschoen.beatplanner.MetroApp;
import dk.aschoen.beatplanner.R;
import dk.aschoen.beatplanner.core.Metronome;

public class MetroActivity  extends Activity 
{
    private Metronome metro;
    private SoundPool sp;
    private int claveId, claveLowId;
    Button startBtn;
    Button stopBtn;
    EditText bpmText;
    SeekBar bpmSeekBar;
    Button bpmIncBtn;
    Button bpmDecBtn;
    Spinner lowerMeterSpinner;
    Spinner upperMeterSpinner;

    protected void onCreate(Bundle bundle) {
        super.onCreate(bundle);
        setContentView(R.layout.activity_metro);
        init();
    }

    private void init() {
        metro = MetroApp.defaultMetronome(this);
        metro.onBPMChangedEvent(new Metronome.OnBPMChangedListener() {
            @Override
            public void onBPMChanged(int oldBPM, int newBPM) {
                bpmText.setText(String.valueOf(newBPM));
                bpmSeekBar.setProgress(newBPM);
            }
        });
        // UI init
        startBtn = (Button) findViewById(R.id.startBtn);
        stopBtn = (Button) findViewById(R.id.stopBtn);
        bpmText = (EditText) findViewById(R.id.bpmText);
        bpmSeekBar = (SeekBar) findViewById(R.id.bpmSeekBar);
        bpmIncBtn = (Button) findViewById(R.id.inc_tempo);
        bpmDecBtn = (Button) findViewById(R.id.dec_tempo);
        upperMeterSpinner = (Spinner) findViewById(R.id.upperMeter);
        lowerMeterSpinner = (Spinner) findViewById(R.id.lowerMeter);
        // Initialize UI with values
        bpmText.setText(String.valueOf(metro.getBPM()));
        bpmSeekBar.setProgress(metro.getBPM());



        upperMeterSpinner.setSelection(3);
        lowerMeterSpinner.setSelection(1);

        upperMeterSpinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
            @Override
            public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
                metro.getBeat().getMeter().setUpper(Integer.valueOf(parent.getItemAtPosition(position).toString()));
            }

            @Override
            public void onNothingSelected(AdapterView<?> parent) {

            }
        });

        lowerMeterSpinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
            @Override
            public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
                metro.getBeat().getMeter().setLower(Integer.valueOf(parent.getItemAtPosition(position).toString()));
            }

            @Override
            public void onNothingSelected(AdapterView<?> parent) {

            }
        });


        startBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                metro.start();
            }
        });


        stopBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                metro.stop();
            }
        });


        bpmIncBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                metro.setBPM(metro.getBPM() + 5);
            }
        });


        bpmDecBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                metro.setBPM(metro.getBPM() - 5);
            }
        });

        bpmText.setOnFocusChangeListener(new View.OnFocusChangeListener() {
            @Override
            public void onFocusChange(View v, boolean hasFocus) {
                try {
                    if (hasFocus == false)
                        metro.setBPM(Integer.valueOf(bpmText.getText().toString()));
                } catch (NumberFormatException e) {
                    e.printStackTrace();
                }
            }
        });

        bpmSeekBar.setOnSeekBarChangeListener(new SeekBar.OnSeekBarChangeListener() {
            @Override
            public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
                metro.setBPM(progress);
            }

            @Override
            public void onStartTrackingTouch(SeekBar seekBar) {

            }

            @Override
            public void onStopTrackingTouch(SeekBar seekBar) {

            }
        });

    }


}


